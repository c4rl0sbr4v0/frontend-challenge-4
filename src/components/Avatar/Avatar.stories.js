import React from 'react';
import Avatar from './index';

export default {
  title: 'Components/Avatar',
  component: Avatar,
};

export const Default = () => <Avatar style={{ display: 'block' }} />;
