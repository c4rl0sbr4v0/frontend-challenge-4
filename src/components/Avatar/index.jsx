import React from 'react';
import r2 from '../../assets/r2.jpg';

/**
 *  Avatar: Render an Avatar
 */
const Avatar = () => {
  return (
    <figure className="media-left landbot-message-avatar">
      <p className="image is-64x64">
        <img className="is-rounded" src={r2} alt="" />
      </p>
    </figure>
  );
};

export default React.memo(Avatar, () => true);
