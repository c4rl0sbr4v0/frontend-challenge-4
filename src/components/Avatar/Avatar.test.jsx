import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Avatar from '.';

test('Avatar Component', async () => {
  const { container } = render(<Avatar />);
  expect(container.firstChild.classList.contains('landbot-message-avatar')).toBe(true);
  expect(container).toMatchSnapshot();
});
