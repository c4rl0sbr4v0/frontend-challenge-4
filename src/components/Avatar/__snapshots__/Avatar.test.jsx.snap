// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component 1`] = `
<div>
  <figure
    class="media-left landbot-message-avatar"
  >
    <p
      class="image is-64x64"
    >
      <img
        alt=""
        class="is-rounded"
        src="r2.jpg"
      />
    </p>
  </figure>
</div>
`;
